### DISCUSSION TITLE
Apply data recovery migration after seeding

### DISCUSSION THREAD

**Discussion by robin-ln** [prisma/prisma#17834] [Q&A]
I work on application where i need to apply a migration after the seedings.

My use case is : 
- i add rows in my table ProjectSetting
- the new projects will be created with the good settings, but i need to apply a migration for update the old projects

In my ci the deploy stage does this : 
-> migrate -> seed

And i need this : 
-> migrate -> seed -> migrate recovery old data

I don't want to setup a seeding that handles data recovery, because after it will become a dead code. My data recovery needs to happen only once (like schema migrations).

I want to use the prisma migration system for play the migration juste one time.

what I imagine : 
```
├ migration
├ migration_custom <----------- custom migration dir
└ schema.prisma
```

and my deploy command
```
prisma migrate
prisma seed
prisma migrate --dir=./migration_custom <----------- new option  
```

Or 

```
├ migration
   └ 123123123_name
        └ migration.sql
        └ data-migration.sql
└ schema.prisma
```
```
prisma migrate <----------- by default --filter=migration.sql 
prisma seed
prisma migrate --filter=data-migration.sql <----------- new option  
```

**Comment by nurul3101** [comment]
Hey @Robin-Ln 👋 

After seeding, I would recommend you to create a new migration with `--create-only` flag, so invoking `npx prisma migrate --create-only` will only create a migration file but will not apply it. You can then edit this empty generated migration file and add the SQL queries for data recovery, then you can again invoke `npx prisma migrate dev` command and that will apply this migration file that you have edited. ([Reference](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations#how-to-edit-a-migration-file))

Would this approach work for you?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
